client/src/styles/shared.ts

```diff
@@ -0,0 +1,51 @@
+
+import * as ToggleGroup from "@radix-ui/react-toggle-group";
+import { colors } from "../colors";
+import styled from "@emotion/styled";
+
+/*
+	I thought why not move out declared styles outside the components, so that they can be reused later on.
+	Tbh, I am not a big fan of styled components, why not css modules or just plain css?
```

Comment:
I don't think the first 3 should be viewed as sharable components, and the 4th (toggle) is arguable, as it's used for only a single debug operation, so moving them out of their respective files can make it harder to understand what is going on inside some of said files.

I think there are a lot of online discussions about the merits or demerits of css-in-js. Personally, I like it, but prefer the 

```ts
const Container = styled.div(css({
  fontFamily: 'Outfit'
}))
```

approach would provides type safety (someone else liked this one and used this, so I just went along).  But as to the benefits, if you have the styling in the same file, it makes it easier to understand everything - logic and styling - that is going on inside that file.  For example, if I have some issue with the AppContainer, where will I look?  Probably in App.tsx, because I want to see how it's being used.  If there is some style that looks weird, I think the best dev experience is to have that logic right there in-line, without having to go seeking for a definition of this object to find the styling.

Things like Tailwind achieve the same thing by having your styling be encapsulated in mostly predefined classes, but still are placed on components inline - you don't need to have `<MyComponent className="myAggregateTailwindClasses" />`, I think for the same reason, in that you can't see when looking at the component what the styling is immediately.